lib/Sema/CSDiag.cpp

```diff
@@ -6686,8 +6686,14 @@ static bool diagnoseKeyPathComponents(ConstraintSystem &CS, KeyPathExpr *KPE,
                                corrections);
 
       if (currentType)
-        TC.diagnose(componentNameLoc, diag::could_not_find_type_member,
-                    currentType, componentName);
+        if (isa<TupleType>(currentType.getPointer())) {
```

Comment:
Aha, this still isn't quite the solution you want: it doesn't look through typealiases.

```
typealias X = (Int, Int)

struct Foo {
  let x: X
}

let kp = \Foo.x.1
```

Instead, you can use `currentType->is<TupleType>()`, which will look through typealiases and other forms of "sugar".

Reply:
That's what I did initially, but @xedin said I do `isa<TupleType>(currentType.getPointer())`. I'll create a new PR to fix this!

Reply:
Sorry if my comment was misleading, I didn’t mean you have to use `isa`, but rather that it’s a one way to check without skipping sugar...

Reply:
Ah okay, I understand now 😄 Thanks for clarifying @xedin!

Reply:
Okay :) @xedin